src/com/google/enterprise/adaptor/filenet/DocumentTraverser.java

```diff
@@ -348,25 +368,23 @@ private void processInheritedPermissions(DocId docId,
 
   private Acl createAcl(DocId docId, Permissions.Acl permissions,
       PermissionSource permSrc, String parentFragment) {
-    Set<String> allowUsers = permissions.getAllowUsers(permSrc);
-    Set<String> denyUsers = permissions.getDenyUsers(permSrc);
-    Set<String> allowGroups = permissions.getAllowGroups(permSrc);
-    Set<String> denyGroups = permissions.getDenyGroups(permSrc);
-    if (allowUsers.isEmpty() && denyUsers.isEmpty()
-        && allowGroups.isEmpty() && denyGroups.isEmpty()) {
-      return null;
-    } else {
-      return createAcl(docId, parentFragment,
-          Acl.InheritanceType.CHILD_OVERRIDES,
-          options.getGlobalNamespace(), allowUsers, denyUsers,
-          allowGroups, denyGroups);
-    }
+    return createAcl(docId, parentFragment,
+        Acl.InheritanceType.CHILD_OVERRIDES,
```

Comment:
That doesn't work.  The only cases that can and do use this method only need CHILD_OVERRIDES.
The LEAF node passes in a union for each set.  The markings do not have a PermissionSource.